hippylib/algorithms/randomizedSVD.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2016-2018, The University of Texas at Austin 
+# & University of California--Merced.
+# Copyright (c) 2019, The University of Texas at Austin 
+# University of California--Merced, Washington University in St. Louis.
+#
+# All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the hIPPYlib library. For more information and source code
+# availability see https://hippylib.github.io.
+#
+# hIPPYlib is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License (as published by the Free
+# Software Foundation) version 2.0 dated June 1991.
+
+import dolfin as dl
+from .multivector import MultiVector, MatMvMult, MatMvTranspmult, MvDSmatMult
+from ..utils import experimental
+import numpy as np
+from scipy.linalg import solve_sylvester
+"""
+Randomized algorithms for the solution of singular value decomposition problem (SVD)
+
+REFERENCES:
+
+Nathan Halko, Per Gunnar Martinsson, and Joel A. Tropp,
+Finding structure with randomness:
+Probabilistic algorithms for constructing approximate matrix decompositions,
+SIAM Review, 53 (2011), pp. 217-288.
+
+Per Gunnar Martinsson
+Randomized Methods for Matrix Computations
+https://arxiv.org/pdf/1607.01649v3.pdf
+"""
+
+def accuracyEnhancedSVD(A,Omega,k,s=1,check=False):
+    """
+    The accuracy enhanced randomized singular value decomposition from  [2].
+    
+    Inputs:
+
+    - :code:`A`: the m x n rectangular operator for which we need to estimate the dominant left-right singular vector / value triplets.
+    - :code:`Omega`: a random gassian matrix with :math:`m \\geq k` columns.
+    - :code:`k`: the number of eigenpairs to extract.
+    
+    Outputs:
+    - :code:`U`: the estimate of the :math:`k` dominant left singular vectors of of :math:`A,\\, U^T U = I_k`.
+    - :code:`sigma`: the estimate of the :math:`k` dominant singular values of :math:`A`.
+    - :code:`V`: the estimate of the :math:`k` dominant right singular vectors of :math:`A,\\, V^T V = I_k`.
+    
+    """
+
+    # Check compatibility of operator A
+    assert hasattr(A,'transpmult'), 'Operator A must have member function transpmult'
+
+    nvec  = Omega.nvec()
+    assert(nvec >= k )
```

Comment:
Not a big deal at all, but maybe you should get rid of parentheses (and add a message)  for consistency purposes.